cli/admin/src/main/java/io/pravega/cli/admin/segmentstore/storage/StorageUpdateSnapshotCommand.java

```diff
@@ -0,0 +1,229 @@
+                return Long.valueOf(file1Epoch).compareTo(Long.valueOf(file2Epoch));
+            }
+            file1 = f1.getName().split(OFFSET_SPLITTER);
+            file2 = f2.getName().split(OFFSET_SPLITTER);
+            return Long.valueOf(file1[file1.length - 1].split("\\.")[0]).compareTo(Long.valueOf(file2[file2.length - 1].split("\\.")[0]));
+        }
+    }
+
+    /**
+     * Update the system snapshot record with the segment chunk details.
+     *
+     * @param systemSnapshot The snapshot to be updated
+     * @param chunkFiles     the segment chunks whose properties need to
+     *                       be updated in the passed system snapshot.
+     */
+    private void updateSystemSnapShotRecord(SystemJournal.SystemSnapshotRecord systemSnapshot, List<File> chunkFiles) {
+        Collection<SystemJournal.SegmentSnapshotRecord> records = systemSnapshot.getSegmentSnapshotRecords();
+        List<ChunkMetadata> chunks = generateChunks(chunkFiles);
+        SystemJournal.SegmentSnapshotRecord recordToBeSaved;
+        Iterator<SystemJournal.SegmentSnapshotRecord> segmentIterator = records.iterator();
+        while (segmentIterator.hasNext()) {
+            SystemJournal.SegmentSnapshotRecord record = segmentIterator.next();
+            if (isSegmentBeingEdited(record.getSegmentMetadata().getName(), chunkFiles.get(0).getName())) {
+                output("Updating SystemSnapshot with relevant data.");
+                record.getSegmentMetadata().setLength(SEGMENT_LENGTH.get());
+                record.getSegmentMetadata().setChunkCount(CHUNK_COUNT.get());
+                record.getSegmentMetadata().setStartOffset(0);
+                record.getSegmentMetadata().setFirstChunk(chunks.get(0).getName());
+                record.getSegmentMetadata().setFirstChunkStartOffset(deriveStartOffset(chunks.get(0).getName()));
+                record.getSegmentMetadata().setLastChunk(chunks.get(chunks.size() - 1).getName());
+                record.getSegmentMetadata().setLastChunkStartOffset(deriveStartOffset(chunks.get(chunks.size() - 1).getName()));
+                record.getSegmentMetadata().setLastModified(System.currentTimeMillis());
+                recordToBeSaved = record.toBuilder().segmentMetadata(record.getSegmentMetadata()).chunkMetadataCollection(chunks).build();
+                segmentIterator.remove();
+                records.add(recordToBeSaved);
+                break;
+            }
+        }
+    }
+
+    /**
+     * Helps determine the segment pointed to by the chunks refered in this command
+     * is the same as the one we have from journal file.
+     */
+    private boolean isSegmentBeingEdited(String segmentFromJournal, String segmentFromChunks) {
+        String[] segmentParts = segmentFromJournal.split("\\/");
+        segmentFromJournal = segmentParts[segmentParts.length - 1];
+        if (segmentFromJournal.split(EPOCH_SPLITTER)[0].equalsIgnoreCase(segmentFromChunks.split(EPOCH_SPLITTER)[0])) {
```

Comment:
You can just return `segmentFromJournal.split(EPOCH_SPLITTER)[0].equalsIgnoreCase(segmentFromChunks.split(EPOCH_SPLITTER)[0])`, as it is analogous as what you are doing here, right?

Reply:
right. changed.